src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -75,9 +75,12 @@ CreateOffer::preflight(PreflightContext const& ctx)
         return temBAD_EXPIRATION;
     }
 
-    bool const bHaveCancel(tx.isFieldPresent(sfOfferSequence));
+    auto const cancelSequence = tx[~sfOfferSequence];
 
-    if (bHaveCancel && (tx.getFieldU32(sfOfferSequence) == 0))
+    if (cancelSequence &&
+        (*cancelSequence == 0 ||
+         (ctx.rules.enabled(fixOfferSequence) &&
+          tx.getSeqProxy() <= SeqProxy::sequence(*cancelSequence))))
```

Comment:
I'm okay with, but not excited about, the extra checks provided by the `fixOfferSequence` amendment.  The checks have a low compute cost and that's great.  I'm bugged that the extra checks are on an amendment.  Because this is a low-benefit amendment, the amendment needs to be postponed so it comes on line at a time when it won't cause servers to become amendment blocked.

If we decide that the change can be safely made without an amendment I'd be fine with that.  Otherwise we need to decide whether the overhead of managing when the amendment becomes active is justified by the (in my opinion) relatively small benefit that the amendment provides.

Reply:
Would it be worth covering this case in the new unit test? Here is a [branch](https://github.com/miguelportilla/rippled/commit/85a58fb086ea88f53da2cf68ac7e0ea3fa06ae9e) checking for the bad sequence.

Reply:
After some further discussions and examination, this new amendment is really just controlling a minor optimization, and really isn't worth it.